doc/design/python-api-1.md

```diff
@@ -0,0 +1,189 @@
+Designs for an improved Python API
```

Comment:
note, in https://github.com/dandi/dandi-cli/pull/712 I renamed `doc/` to `docs/` so if we are to merge this one, might want to rename